app/models/piece.rb

```diff
@@ -1,3 +1,130 @@
 class Piece < ApplicationRecord
   belongs_to :game
+
+  # anywhere it say y_coord was self.y & x_coord was self.x
+
+                # Need function to automoatically keep up with current pieces coords at all times.
+  def current_pieces_coords    # Will need to be removed later after new function made.  
+    [[0,7],[3,7],[4,7],[5,7],[6,7],[7,7],
+    [2,6],[3,6],[4,6],[5,6],[6,6],[7,6],
+    [0,5],[1,5],[7,5],
+    [2,3],[3,3],
+    [0,1],[1,1],[4,1],[5,1],[6,1],[7,1],
+    [0,0],[1,0],[3,0],[5,0],[6,0],[7,0]]      # Represents all the coordinates of pieces in play
+  end
+
+  def is_obstructed?(x, y)
+    if x_coord == x
+      is_vertically_obstructed?(x,y)
+    elsif y_coord == y 
+      is_horizontally_obstructed?(x,y)
+    elsif invalid_input?(x,y) == false
+      is_diagonally_obstructed?(x,y)
+    else
+      invalid_input?(x,y)
+    end
+  end
+
+  def is_vertically_obstructed?(x, y) 
+    y_range = []
+    if y_coord < y
+      (y_coord..y).each { |n| y_range << n }   # If statement catches if numbers go high to low
+    else
+      (y..y_coord).each { |n| y_range << n }   # Range doesn't work backwards
+      y_range = y_range.reverse               # This line reverses it back to the correct order
+    end
+    y_range.pop         # removes first number
```

Comment:
Maybe we can omit pop/shift by starting our range with `y+1` and ending in `y_coord -1`?